File: Jetpack/src/main/java/com/example/jetpack/dao/BookDao.java
package com.example.jetpack.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import com.example.jetpack.entity.Book;

import java.util.List;

@Dao
public interface BookDao {

    @Insert
    Long insertBook(Book book);

    @Query("select * from Book")
    List<Book> loadAllBooks();
}

File: Kotlin/src/main/java/com/example/learnKotlin/chapter7/dao/BookDao.kt
package com.example.learnKotlin.chapter7.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query
import com.example.learnKotlin.chapter7.entity.Book

@Dao
interface BookDao {

    @Insert
    fun insertBook(book: Book): Long

    @Query("select * from Book")
    fun loadAllBooks(): List<Book>
}